server/apps/publish/archive_publish.py

```diff
@@ -282,48 +302,92 @@ def publish(self, doc, updates, target_output_channels=None):
                 updates['source'] = desk['source'] if desk and desk.get('source', '') \
                     else DEFAULT_SOURCE_VALUE_FOR_MANUAL_ARTICLES
 
+            updates['pubstatus'] = 'canceled' if self.publish_type == 'killed' else 'usable'
```

Comment:
Would it make sense to define the list of possible `pubstatus` states in some kind of enum-ish object and use that instead of plain strings? Would provide a better protection against typos etc.